editor.js

```diff
@@ -681,11 +681,15 @@ function uploadMergeCode(append) {
         }
 
         var newCode = uploadedXML;
-        newCode = newCode.substring(42, newCode.length);
+        if (newCode.indexOf('<variables>') > -1) {
+            newCode = newCode.substring(uploadedXML.indexOf('<block'), newCode.length);
+        } else {
+            newCode = newCode.substring(uploadedXML.indexOf('<variables>'), newCode.length);
+        }
         newCode = newCode.substring(0, (newCode.length - 6));
```

Comment:
This is minor, but the number '6' should be a constant to clue developers into what (newCode.length - 6) really represents.